src/Bicep.LangServer/Completions/BicepCompletionProvider.cs

```diff
@@ -1244,8 +1256,17 @@ private static CompletionItem CreateSymbolCompletion(Symbol symbol, Range replac
                     completion.WithCommand(new Command { Name = EditorCommands.SignatureHelp });
                 }
 
+                ImmutableArray<FunctionOverload> overloads = function.Overloads;
+                var genericDescription = overloads.First().GenericDescription;
+                string codeBlock = string.Empty;
+
+                foreach (var overload in overloads)
+                {
+                    codeBlock = codeBlock + function.Name + overload.TypeSignature + "\n";
```

Comment:
>codeBlock + function.Name + overload.TypeSignature + "\n"; [](http://example.com/codeflow?start=32&length=58)

You should use string interpolation expressions instead of manually concatenating strings. Strings are immutable in .net so every + may potentially allocate a new string unless the compiler optimizes the whole thing away. Separately from that .net 6 made significant improvements in optimizations on string interpolation expressions.

Reply:
Another option would be to use a `StringBuilder` since `codeBlock` is constructed inside a loop.

Reply:
Great point, thanks!